### PR TITLE
Messages Pallet Documentation Improvement

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to improve the documentation of the Pallets and make that documentation be able to be used on docs.frequency.xyz.

Part of https://github.com/frequency-chain/docs/issues/59

# Discussion
- Messages Pallet Readme

# Screenshot
image

### CODE REVIEW

File: pallets/messages/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Messages stores metadata and message payload data for a set Schema.
+Use of the Message pallet payloads is designed for time-centric data.
+Discovery is via the `MessagesInBlock` on new blocks or requesting messages from a block.
+Retrieval is via RPC.
+
+### Metadata vs Payload
+
+Messages have both metadata and payloads.
+The payload should always match the data structure or the messages is considered invalid.
```
Comment (jeanettedepatie) [inline pallets/messages/README.md:15]: Should read "or the message is considered invalid" or "or the messages are considered invalid"

File: pallets/messages/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+The metadata is the Block Number, Schema Id, and other data to help discover and organize the payload information.
```
Comment (jeanettedepatie) [inline pallets/messages/README.md:16]: Would suggest "and other data useful for discovering and organizing the payload information".

File: pallets/messages/README.md (hunk 1 above)
Comment (wilwade) [reply pallets/messages/README.md:15]:
```suggestion
The payload should always match the data structure or the message is considered invalid.
```

File: pallets/messages/README.md (hunk 2 above)
Comment (wilwade) [reply pallets/messages/README.md:16]:
```suggestion
The metadata is the Block Number, Schema Id, and other data useful for discovering and organizing the payload information.
```

File: pallets/messages/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+### Metadata vs Payload
+
+Messages have both metadata and payloads.
+The payload should always match the data structure or the message is considered invalid.
+The metadata is the Block Number, Schema Id, and other data useful for discovering and organizing the payload information.
+
+### Payload Options
+
+- `IPFS`: Storage of the CID and length of the file on IFPS
```
Comment (JoeCap08055) [inline pallets/messages/README.md:20]:
```suggestion
- `IPFS`: Storage of the CID and length of the file on IPFS
```

File: pallets/messages/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+- `OnChain`: Storage of the entire payload data. Usually for tiny payloads only.
+
+### Message Ordered
+
+Messages are ordered by block number, but the have a set order based on transaction order in the block.
```
Comment (JoeCap08055) [inline pallets/messages/README.md:25]:
```suggestion
### Message Ordering

Messages are ordered by block number, and within each block have a set order based on transaction order in the block.
```

File: pallets/messages/src/lib.rs [hunk 5]
```diff
@@ -1,37 +1,12 @@
-//! # Messages pallet
-//! A pallet for storing messages.
+//! Stores messages `IPFS` and `OnChain` Schema payload locations
```
Comment (JoeCap08055) [inline pallets/messages/src/lib.rs:1]:
```suggestion
//! Stores messages for `IPFS` and `OnChain` Schema payload locations
```

File: pallets/messages/src/lib.rs [hunk 6]
```diff
@@ -1,37 +1,12 @@
-//! # Messages pallet
-//! A pallet for storing messages.
+//! Stores messages for `IPFS` and `OnChain` Schema payload locations
 //!
```
Comment (wilwade) [inline pallets/messages/src/lib.rs:2]:
```suggestion
//!
//! ## Quick Links
```

File: pallets/messages/README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+### Metadata vs Payload
+
+Messages have both metadata and payloads.
+The payload should always match the data structure or the message is considered invalid.
+The metadata is the Block Number, Schema Id, and other data useful for discovering and organizing the payload information.
+
+### Payload Options
+
+- `IPFS`: Storage of the CID and length of the file on IPFS
```
Comment (enddynayn) [inline pallets/messages/README.md:20]: period

File: pallets/messages/README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+- `OnChain`: Storage of the entire payload data. Usually for tiny payloads only.
```
Comment (enddynayn) [inline pallets/messages/README.md:21]: not a big deal but what does tiny mean?

File: pallets/messages/README.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+### Message Ordering
+
+Messages are ordered by block number, and within each block have a set order based on transaction order in the block.
```
Comment (enddynayn) [inline pallets/messages/README.md:25]: Maybe:
Messages are ordered by block number, and within each block, they follow a specific order based on their transaction sequence within that block.

File: pallets/messages/README.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+This order is immutable.
+
+### Actions
+
+The Messages pallet provides for:
+
+- Adding messages for a given schema
```
Comment (enddynayn) [inline pallets/messages/README.md:32]: minor: but sometimes there is a period at the end of these and sometimes there is not.

**Review by enddynayn — APPROVED** [review]
great 👍

**Review by mattheworris — APPROVED** [review]
Lookin' good, 🚢 it!

File: pallets/messages/README.md (hunk 8 above)
Comment (wilwade) [reply pallets/messages/README.md:21]: Updated to:
`- `OnChain`: Storage of the entire payload data, usually for sub-256 byte payloads`

File: pallets/messages/README.md (hunk 7 above)
Comment (wilwade) [reply pallets/messages/README.md:20]: Removing periods elsewhere. I've been going without most places, and the one below slipped through with the original statement having two parts.

File: pallets/messages/README.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
+# Messages Pallet
+
+Stores block-indexed data for a Schema using `OnChain` or `IPFS` payload location.
+
+## Summary
+
+Messages stores metadata and message payload data for a set Schema.
+Use of the Message pallet payloads is designed for time-centric data.
```
[the remark this answers is not in the document]
Comment (wilwade) [reply pallets/messages/README.md]: Merging for now, but let me know if you have thoughts/better ideas and we can tweak it in one of the other docs PRs.

File: pallets/messages/README.md (hunk 8 above)
Comment (wilwade) [reply pallets/messages/README.md]: Merging for now, but let me know if you have thoughts/better ideas and we can tweak it in one of the other docs PRs.

**Pull request merged**